src/environmentd/src/http/webhook.rs

```diff
@@ -55,6 +61,57 @@ pub async fn handle_webhook(
     Ok(())
 }
 
+async fn validate_request(
+    validate_expr: AppendWebhookValidation,
+    body: Bytes,
+    headers: &http::HeaderMap,
+) -> Result<(), WebhookError> {
+    // Convert our headers into a map for validation.
+    let mut headers_s = BTreeMap::new();
+    for (name, val) in headers.iter() {
+        if let Ok(val_s) = val.to_str() {
+            headers_s.insert(name.as_str().to_string(), val_s.to_string());
+        }
+    }
+
+    // Run our validation on a separate thread, waiting for it's completion.
+    let valid = mz_ore::task::spawn_blocking(
+        || "webhook-validate".to_string(),
+        move || {
+            let mut row = Row::with_capacity(2);
+            let mut packer = row.packer();
+
+            packer.push(Datum::Bytes(&body[..]));
+            packer.push_dict(
+                headers_s
+                    .iter()
+                    .map(|(name, val)| (name.as_str(), Datum::String(val))),
+            );
+
+            // TODO(parkmycar): Is there a way to get a Datum::Map without packing and unpacking a
+            // row?
+            let datums = row.unpack();
+            let body = datums.get(0).copied().unwrap_or(Datum::Null);
+            let headers = datums.get(1).copied().unwrap_or(Datum::Null);
+
+            // Since the validation expression is technically a user defined function, we want to
+            // be extra careful and guard against issues taking down the entire process.
+            mz_ore::panic::catch_unwind(move || validate_expr(body, headers))
```

Comment:
Did you encounter panics here during testing? IMO if this panics we've done something quite wrong on our end. Kinda think it's better for it to panic loudly?

Reply:
I did not encounter this, but the reason I wanted to be extra careful is because from the user perspective, I think webhook sources are more opaque than others. For example, with a Kafka or Postgres source the user also controls those components, and if they're behaving badly they can debug them. AFAIK for webhooks there is limited debugging/visibility into the service pushing us data, so if something goes wrong I think it'll be harder to track down, which is why I didn't want the consequence of bad validation to panic all of `environmentd`.

Although I could be over thinking this, more than happy to change :)

Reply:
As long as it loudly shows up in our infra. It's ok to keep the unwind and send an error to the user (at that point it should be an internal error), but we must also log to sentry with `error!` to ensure we see it.

Reply:
Sounds good to me! `ValidationPanicked` returns to the user as a 500 and we use `tracing::error` so it shows up in our infra